### PR TITLE
Ogre 1.12.10 upgrade

### PR DESCRIPTION
**Pull request by ijnek**
Continuing on from https://github.com/ros2/rviz/tree/clalancette/ogre-1.12.10-upgrade

The only thing I wasn't sure how to test was the ViewPicker. I'm not quite sure what that does and how to test it.

### CODE REVIEW

File: rviz_ogre_vendor/package.xml
Comment (clalancette) [inline rviz_ogre_vendor/package.xml]: Adding this dependency is unfortunate, because we'll need to package it for Windows.

But I'm actually wondering if we can avoid it completely.  It looks like only 2 things in OGRE use it; the XMLConverter and the DotSceneLoader.  I don't think either of those is required for RViz, so we could get away with changing the CMakeLists.txt to have:

```
list(APPEND extra_cmake_args "-DOGRE_BUILD_TOOLS:BOOL=OFF")
list(APPEND extra_cmake_args "-DOGRE_BUILD_PLUGIN_DOT_SCENE:BOOL=OFF")
```

**Review by clalancette — CHANGES_REQUESTED** [review]
Thanks for picking up this work and doing additional fixing and testing!

I've got one thing I think needs to be changed.  Otherwise, I'm overall pretty happy with this, though I'd really appreciate if one of the rviz2 maintainers could review and test it as well.

File: rviz_ogre_vendor/package.xml
Comment (ijnek) [reply rviz_ogre_vendor/package.xml]: Changes have been made

**Review by clalancette — APPROVED** [review]
This looks good to me, thanks for pushing this over the line.  I'll run CI on it to see if that comes up with anything.

@jacobperron @mjeronimo Any thoughts on this?  Maybe if you have a chance, you can take it for a spin.

**Comment by clalancette** [comment]
CI:
* Linux [Build Status](http://ci.ros2.org/job/ci_linux/16977/)
* Linux-aarch64 [Build Status](http://ci.ros2.org/job/ci_linux-aarch64/11542/)
* Windows [Build Status](http://ci.ros2.org/job/ci_windows/17381/)

**Comment by clalancette** [comment]
@ijnek It looks like there are a couple of failing tests on Linux, and a compiler warning on Windows.  Can you take a look?  Thanks.

File: rviz_common/include/rviz_common/frame_position_tracking_view_controller.hpp
Comment (jacobperron) [inline rviz_common/include/rviz_common/frame_position_tracking_view_controller.hpp]: I'm not sure why the deprecation notice recommends including `Ogre.h`, but I think it'd be better to be narrow in our includes if possible. I _think_ the header we want is `<OgreVector.h>` ([reference](https://github.com/OGRECave/ogre/blob/v1.12.13/OgreMain/include/OgreVector3.h#L1)).

Same comment for other places where we replace OgreVector3.h with Ogre.h

**Review by jacobperron — COMMENTED** [review]
Changes look okay to me. Just one comment.

File: rviz_common/include/rviz_common/frame_position_tracking_view_controller.hpp
Comment (ijnek) [reply rviz_common/include/rviz_common/frame_position_tracking_view_controller.hpp]: I agree, I'll give the changes a try.

**Comment by ijnek** [comment]
@clalancette I'll have a look now.

---

Update

I believe the failing tests are due to some bugs in upstream ogre introduced in their 1.12.10 release (the tests pass in 1.12.9). I'm going to see if I can isolate and confirm if it is a bug and make a fix upstream. Nonetheless, ogre doesn't seem to backport such bug fixes so we'll either have to disable the test, or modify the test to get around the bug...

**Comment by ijnek** [comment]
The bug was fixed upstream, but looks like we won't get the changes backported to Ogre1.12. I think modifying the test to work around the bug is the best way around as it only seems to affect the tests.

**Comment by ijnek** [comment]
@ros-pull-request-builder retest this please

**Comment by ijnek** [comment]
In regards to the windows build failing - I set up a windows VM locally (please tell me if there's an easier way to test these changes) but couldn't reproduce the issue. Let's see if it happens again...

**Comment by ijnek** [comment]
> @ros-pull-request-builder retest this please

How do I trigger the tests correctly?

**Review by jacobperron — APPROVED** [review]
LGTM, pending CI

**Comment by jacobperron** [comment]
These tests are manually triggered by the ROS 2 team.

* Linux [Build Status](http://ci.ros2.org/job/ci_linux/17014/)
* Linux-aarch64 [Build Status](http://ci.ros2.org/job/ci_linux-aarch64/11575/)
* Windows [Build Status](http://ci.ros2.org/job/ci_windows/17430/)

**Comment by jacobperron** [comment]
There's a MSBuild warning coming from `OgreRenderSystem.h:983`,

```
'argument': conversion from 'size_t' to 'const T', possible loss of data
```

I'm not sure this is something we can fix in RViz, or if we should just suppress it with some preprocessor directives.

**Comment by clalancette** [comment]
> I'm not sure this is something we can fix in RViz, or if we should just suppress it with some preprocessor directives.

I suspect (but I'm not sure) that it has to do with the argument we are passing in when we actually call one of the Ogre functions.  If that is the case, we can probably fix it in RViz, but it should be investigated one way or another.

That said, this is really close!  Once we fix that issue one way or another, I think we'll be ready for this to land.

**Comment by ijnek** [comment]
The issue seems to be coming from a deprecated method in Ogre that isn't used anywhere, so I just added a patch (C4267.patch) that comments it out. C4267 is the exact windows error code.
In upstream, the issue was fixed in https://github.com/OGRECave/ogre/commit/ba07f9616ad43d43469b50c062c94674338b2aa so it is included in 13.0.0 onwards. This patch can be deleted once we reach 13.0.0.

This is ready for CI to be run again.

**Comment by ijnek** [comment]
@clalancette or @jacobperron Could one of you run the CI again please?

**Comment by clalancette** [comment]
Another CI:
* Linux [Build Status](http://ci.ros2.org/job/ci_linux/17060/)
* Linux-aarch64 [Build Status](http://ci.ros2.org/job/ci_linux-aarch64/11622/)
* Windows [Build Status](http://ci.ros2.org/job/ci_windows/17487/)

**Comment by clalancette** [comment]
CI was happy here, and I did some local testing just to ensure that rviz worked properly.  Going ahead and merging, thanks for all of the work here.

**Pull request merged**
